openmetadata-service/src/main/java/org/openmetadata/service/search/SearchClient.java

```diff
@@ -68,7 +68,8 @@ public interface SearchClient {
   String SOFT_DELETE_RESTORE_SCRIPT = "ctx._source.put('deleted', '%s')";
   String REMOVE_TAGS_CHILDREN_SCRIPT =
       "for (int i = 0; i < ctx._source.tags.length; i++) { if (ctx._source.tags[i].tagFQN == params.fqn) { ctx._source.tags.remove(i) }}";
-
+  String REMOVE_TAGS_WITH_PARENT_SCRIPT =
```

Comment:
why are we doing this with painless script. You can do this by searchByQuery and find the docs that has the tag and remove that tag only

Reply:
thanks @harshach for the suggestion, have updated the logic using searchByQuery and removed the painless script.

Reply:
we are going with painless script here  as in the tags field we have  no other information about the parent other than in the fullyQualifiedName, 
for example here, the only information for parent that we have is from `tagFQN` ,  that classification name is `departments` -
"tags" : [
            {
              "tagFQN" : "departments.marketing",
              "displayName" : "marketing",
              "name" : "marketing",
              "labelType" : "Manual",
              "description" : "marketing",
              "style" : { },
              "source" : "Classification",
              "state" : "Confirmed"
            }
          ],

Reply:
@sonika-shah look into updateByQuery, painless is non-performant and adds on top of the results fetched. what you need to do is updatedByquery and fetch all the docs that has the tag thats deleted and remove it.